src/app/charts/extra-params-components/threshold.component.ts

```diff
@@ -0,0 +1,77 @@
+import { AfterViewInit, Component, EventEmitter, OnChanges, Input, Output } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+
+import * as _ from 'lodash';
+
+/*
+ * Chart component
+ * Container for each individual chart plus controls
+ */
+@Component({
+  selector: 'ccl-threshold-parameters',
+  templateUrl: './threshold.component.html'
+})
+export class ThresholdComponent implements AfterViewInit, OnChanges {
+
+    @Input() label: string;
+
+    thresholdForm: FormGroup;
+
+    @Input() comparators: any[] = [
+        {'key': 'gte', 'label': 'greater than or equal to'},
+        {'key': 'lte', 'label': 'less than or equal to'},
+        {'key': 'gt', 'label': 'greater than'},
+        {'key': 'lt', 'label': 'less than'}
+    ];
+    @Input() thresholdUnits: any[] = [
+        {'key': 'K', 'label': 'Kelvin'},
```

Comment:
Moved out to separate class variables, could relocate if we need them elsewhere.